packages/dds/task-manager/src/test/taskManager.spec.ts

```diff
@@ -58,163 +58,207 @@ describe("TaskManager", () => {
-            const lockTaskP2 = taskManager2.lockTask(taskId);
+            const volunteerTaskP1 = taskManager1.volunteerForTask(taskId);
+            const volunteerTaskP2 = taskManager2.volunteerForTask(taskId);
 
             assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
-            assert.ok(!taskManager1.haveTaskLock(taskId), "Task manager 1 should not have lock");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
             assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
-            assert.ok(!taskManager2.haveTaskLock(taskId), "Task manager 2 should not have lock");
+            assert.ok(!taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
 
             containerRuntimeFactory.processAllMessages();
-            await lockTaskP1;
+            const isAssigned1 = await volunteerTaskP1;
+            assert.ok(isAssigned1, "Should resolve true");
 
             assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
-            assert.ok(taskManager1.haveTaskLock(taskId), "Task manager 1 does not have lock");
+            assert.ok(taskManager1.assignedTask(taskId), "Task manager 1 is not assigned task");
             assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
-            assert.ok(!taskManager2.haveTaskLock(taskId), "Task manager 2 should not have lock");
+            assert.ok(!taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
 
             taskManager1.abandon(taskId);
             containerRuntimeFactory.processAllMessages();
-            await lockTaskP2;
+            const isAssigned2 = await volunteerTaskP2;
+            assert.ok(isAssigned2, "Should resolve true");
 
             assert.ok(!taskManager1.queued(taskId), "Task manager 1 should be queued");
-            assert.ok(!taskManager1.haveTaskLock(taskId), "Task manager 1 should not have lock");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
             assert.ok(taskManager2.queued(taskId), "Task manager 2 should be queued");
-            assert.ok(taskManager2.haveTaskLock(taskId), "Task manager 2 should not have lock");
+            assert.ok(taskManager2.assignedTask(taskId), "Task manager 2 should not be assigned");
+        });
+
+        it("Can subscribe to a task", async () => {
+            const taskId = "taskId";
+            taskManager1.subscribeToTask(taskId);
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(!taskManager1.assignedTask(taskId), "Task manager 1 should not be assigned");
+
+            containerRuntimeFactory.processAllMessages();
+
+            assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+            assert.ok(taskManager1.assignedTask(taskId), "Task manager 1 is not assigned task");
```

Comment:
Nit: My preference for the assert language is to use "should" statements, which makes it more clear to a first-time reader what the expectation is (vs. "is" statements, it's not immediately clear if you're saying "is but should not" vs. "is expected to").  This is somewhat personal preference though.